packages/pigment-css-react/src/base.d.ts

```diff
@@ -35,3 +35,32 @@ export type CSSObjectNoCallback =
   | CSSPropertiesMultiValue
   | CSSPseudosNoCallback
   | CSSOthersObjectNoCallback;
+
+export type BaseDefaultProps = object;
+
+export type NoInfer<T> = [T][T extends any ? 0 : never];
+type FastOmit<T extends object, U extends string | number | symbol> = {
+  [K in keyof T as K extends U ? never : K]: T[K];
+};
+
+export type Substitute<A extends object, B extends object> = FastOmit<A, keyof B> & B;
```

Comment:
Let's find time to comment on the purpose of these types later 😅. I can imagine how hard to make the `PolymorphicComponent` work.